src/lib/core/a11y/focus-trap.ts

```diff
@@ -11,48 +11,72 @@ import {coerceBooleanProperty} from '../coercion/boolean-property';
  * Things like tabIndex > 0, flex `order`, and shadow roots can cause to two to misalign.
  * This will be replaced with a more intelligent solution before the library is considered stable.
  */
-@Component({
-  moduleId: module.id,
-  selector: 'cdk-focus-trap, focus-trap',
-  templateUrl: 'focus-trap.html',
-  encapsulation: ViewEncapsulation.None,
+@Directive({
+  selector: 'cdk-focus-trap, focus-trap, [cdk-focus-trap], [focus-trap]',
 })
-export class FocusTrap {
-  @ViewChild('trappedContent') trappedContent: ElementRef;
+export class FocusTrap implements AfterViewInit, OnDestroy {
+  private _startAnchor: HTMLElement = this._createAnchor();
+  private _endAnchor: HTMLElement = this._createAnchor();
 
   /** Whether the focus trap is active. */
   @Input()
   get disabled(): boolean { return this._disabled; }
-  set disabled(val: boolean) { this._disabled = coerceBooleanProperty(val); }
+  set disabled(val: boolean) {
```

Comment:
Hmm, I didn't really consider that. My gut feeling is that Angular would just pass the disabled value to both the component and the directive.